test/functional/mempool_reorg.py

```diff
@@ -106,8 +106,45 @@ def run_test(self):
 
         self.log.info("Check that the mempool is empty")
         assert_equal(set(self.nodes[0].getrawmempool()), set())
+        assert_equal(set(self.nodes[1].getrawmempool()), set())
         self.sync_all()
 
 
+        self.log.info("Simulate natural reorg")
+        self.connect_nodes(0, 1)
+        assert_equal(self.nodes[0].getblockcount(), self.nodes[1].getblockcount())
+        assert_equal(set(self.nodes[0].getrawmempool()), set(self.nodes[0].getrawmempool()))
+        self.generate(self.nodes[0], 3)
+        # Create another transaction which is time-locked to two blocks in the future
+        utxo = wallet.get_utxo()
+        timelock_tx_2 = wallet.create_self_transfer(
+            from_node=self.nodes[0],
+            utxo_to_spend=utxo,
```

Comment:
nit: Why is this needed? The wallet will pick a utxo by itself.

Reply:
removed